### PR TITLE
search contexts: add search box to single page view

### LINKED ISSUE #24407: search contexts: add search box to single page view
Add search box to the search contexts single page view with pre-filled current context. Then we need a way of locking the search context dropdown to prevent changing the pre-filled context.

Problem: There is a potential for confusion between a regular full-featured search box and the search notebook query inputs if they appear on the same page. Search box takes you to a separate page once you submit a search, meanwhile search notebooks display the results inline.

### PR DESCRIPTION
**Pull request by novoselrok**
Examples:

Screenshot 2021-08-31 at 08 33 10

Screenshot 2021-08-31 at 08 18 01

Fixes #24407

### CODE REVIEW

**Comment by novoselrok** [comment]
TODO: @quinnkeast We need some way of "locking" the pre-selected search context so users can't change it. I was thinking of just adding a small lock icon next to the context but open to suggestions.

File: client/web/src/searchContexts/SearchContextPage.story.tsx [hunk 1]
```diff
@@ -156,14 +192,16 @@ repo:^github\\.com/sourcegraph/sourcegraph$ type:diff before:yesterday after:"la
 )
 
 add(
-    'autodefined context',
+    'search box with selected context',
```
Comment (limitedmage) [inline client/web/src/searchContexts/SearchContextPage.story.tsx:195]: minor: it seems odd to me that these tests show the page without a selected context, with the exception of this test. I would expect this page to always show the context selector with a selected context.

**Review by limitedmage — APPROVED** [review]

**Comment by benvenker** [comment]
@novoselrok, what state am I in when I see the first screenshot? I'm a little confused about how I would get there.

**Comment by novoselrok** [comment]
> @novoselrok, what state am I in when I see the first screenshot? I'm a little confused about how I would get there.

No special state needed, you would directly navigate to a search context. Either from the `/contexts` page or directly going to `/contexts/public-ctx`.

**Comment by novoselrok** [comment]
> Ah ok. I was expecting the default page to show the notebook components. Are you just showing one example without them and one with?

Yep, that’s just the examples. Notebooks are optional to include in the description.

**Review by benvenker — APPROVED** [review]

**Pull request closed without merging**